src/passes/variableDeclarationExpressionSplitter.ts

```diff
@@ -30,6 +43,90 @@ export class VariableDeclarationExpressionSplitter extends ASTMapper {
   generateNewConstantName(): string {
     return `__warp_td_${this.lastUsedConstantId++}`;
   }
+
+  // Helper function to generate the typeName for a given typeNode, to be used for
+  // the new VariableDeclarationStatement in this.splitDeclaration
+  generateTypeNameNode(
```

Comment:
We already have typeNameFromTypeNode in utils, would be worth checking that they produce the same result, adding any cases this covers that typeNameFromTypeNode doesn't, and then using that